cli/cmd/root.go

```diff
@@ -71,7 +71,20 @@ func NewCmdRoot() *cobra.Command {
 			cmd.SilenceUsage = true
 			var err error
 
-			invocationUUID, err = uuid.NewRandom()
+			uuidInput, err := cmd.Flags().GetString("invocation-uuid")
```

Comment:
A cleaner approach can be to use `PersistentFlags().Var` and add a `UUID` wrapper struct that [implements](https://github.com/spf13/pflag/blob/d5e0c0615acee7028e1e2740a11102313be88de1/flag.go#L187) `pflag.Value` similar to https://github.com/cloudquery/cloudquery/blob/07e176bf8ae9331686547b4f53ab175df04117dc/cli/internal/enum/enum.go#L27

Reply:
That will ensure the error is reported during parsing as soon as possible

Reply:
Implemented in [ac8c968](https://github.com/cloudquery/cloudquery/pull/18878/commits/ac8c96810a2213958a8aeb18d8472017c4169ae5)